File: java/src/test/java/problems201_300/CountCompleteTreeNodesTest.java
package problems201_300;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class CountCompleteTreeNodesTest extends SampledTest {
    @Test
    public void test() {
        for (CountCompleteTreeNodes solution : CountCompleteTreeNodes.values()) {
            for (CountCompleteTreeNodesSample sample
                : getSamples(CountCompleteTreeNodesSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.countNodes(TreeNode.of(sample.input.root)))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/CountCompleteTreeNodesTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class CountCompleteTreeNodesTest : SampledTest() {
    @Test
    fun test() =
        CountCompleteTreeNodes.entries.forEach { solution ->
            getSamples<Array<CountCompleteTreeNodesSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.countNodes(TreeNode.of(*input.root)))
                    .isEqualTo(output)
            }
        }
}
